posts/bayesian-model-comparison/index.qmd

```diff
@@ -0,0 +1,470 @@
+  color="y", 
+  alpha=0.7, 
+  label="P(svm < rf)"
+)
+ax.set_title(
+  "Posterior distribution over the difference in error, RF - SVM. Negative means "
+  "SVM is better.",
+  fontdict={'fontsize': 10, 'fontweight': 'bold'},
+)
+ax.axvspan(*hdi, alpha=0.2, color='gray', label="94% HDI")
+ax.annotate(f" Mean = {round(mean, 3)}", xy=(mean, y_mean + 0.5))
+ax.annotate(f" P(svm > rf) = {round(prob_svm_better, 4)}", xy=(-x_lim, y_mean/2))
+ax.annotate(f" P(svm ~ rf) \n         = \n      {round(prob_rope, 4)}", xy=(ROPE[0], y_mean/2))
+ax.annotate(f" P(svm < rf) = {round(prob_rf_better, 4)}", xy=(mean + 0.75*(hdi[1]-mean), y_mean/2))
+
+ax.legend();
+```
+
+Since the posterior distribution informs us about the relative credibility values across the reals, from the posterior we get the uncertainty in the estimate.
+From this we can get a whole lot of useful information, for instance the Highest Density Intervals (HDIs), the mode, the mean, etc.
+Furthermore, equipped with the posterior distribution and the region of practical equivalence we can:
+
+1) estimate the posterior probability of a reasonable null hypothesis, i.e. if the difference in performance is within a couple of percentage points they may well be considered equivalent. 
+This will be given by the area within the rope region, above denoted by $P(M_1 \sim M_2)$.
+2) significant differences also have practical meaning (area outside the ROPE)
+3) represents effect size and uncertainty (HDI)
+4) we can make meaningful decision based on requiring that the probability of a model being better than another has to be at least 95%, say.
+
+## Cost sensitive decisions
+We will be dealing with this in more detail in a separate blog post. 
+In the real world, it is usually the case that we want to reason about situations and make decisions based on costs and errors. 
+Phrased differently, the choice of the cost measure should depend on how the system is going to be used, rather than on any inherent specification of the training process.
+The issue with doing this is that it is hard. 
+It is hard because specifying a cost-aware loss function is non-trivial, because cost-specifications are domain specific, and becuase even in the case of roughly knowing what the costs are, using this information is hard, i.e. the specified weigthed cost may be a difficult objective for optimisers to work with.
+However if we can specify costs then decision making based on these would be the best way to work as this allows one to take into consideration the utility of the decision that will be made.
+
+In our situation, in order to do cost-sensitive analysis/decision making, all we need is to specify a cost function -- we won't need to run an optimisation on this objective function. 
+This is a function that defines the loss (or cost) we incur in making a given decision (e.g. the wrong decision).
+A typical example is whether to give more importance to a false positive or a false negative.
```

Comment:
wouldn't this be picked up by error/score, ie F1 or otherwise?

Reply:
In some sense, yes. However without specifying exactly what a false negative or false positive costs you you can't really define a good beta. And even if you do that harmonic mean of precision and recall doesn't necessarily become granular enough. Fbeta does go towards that though, yes. 

Maybe I should expand on this/clarify

Reply:
nah. i fear we will be expanding on everything. i get you point. happy to leave as is and resolve conversation